metaflow/plugins/kfp/kfp.py

```diff
@@ -1,86 +1,293 @@
 import kfp
 from kfp import dsl
+from kubernetes.client.models import V1EnvVar
 
-from .constants import DEFAULT_RUN_NAME, DEFAULT_EXPERIMENT_NAME, DEFAULT_FLOW_CODE_URL, DEFAULT_KFP_YAML_OUTPUT_PATH
+from .constants import DEFAULT_FLOW_CODE_URL, DEFAULT_KFP_YAML_OUTPUT_PATH, DEFAULT_DOWNLOADED_FLOW_FILENAME
+from .constants import S3_AWS_ARN as S3_AWS_ARN_VALUE
+from .constants import S3_BUCKET as S3_BUCKET_VALUE
+from typing import NamedTuple
+from collections import deque
 
-def get_ordered_steps(graph):
+StepOutput = NamedTuple('StepOutput', [('ds_root', str), ('run_id', str)])
+
+def step_op_func(python_cmd_template, step_name: str,
+                 code_url: str,
+                 ds_root: str,
+                 run_id: str,
+               ):
     """
-    Returns the ordered step names in the graph (FlowGraph) from start step to end step as a list of strings containing the
-    step names.
+    Function used to create a KFP container op (see: `step_container_op`) that corresponds to a single step in the flow.
 
-    # TODO: Support other Metaflow graphs, as branching and joins are not currently supported
-    Note: All MF graphs start at the "start" step and end with the "end" step (both of which are mandatory).
     """
+    import subprocess
+    import os
+
+    MODIFIED_METAFLOW_URL = 'git+https://github.com/zillow/metaflow.git@state-integ-s3'
+    DEFAULT_DOWNLOADED_FLOW_FILENAME = 'downloaded_flow.py'
```

Comment:
I noted the problem with using imports on the s3 integ PR and also the problem with using the `use_code_pickling` option. So seems best to do it this way. 

So any code changes to these constants will happen only in `constants.py` and inside of the `op_functions`

Reply:
I'm not following, can this not be imported within the function?

Reply:
Yes, the import does not work because of the following: 

The container needs to have Metaflow installed in order for it to be able to resolve imports. But the thing we're trying to import is actually the URL needed to download metaflow. So, I need to pass it explicitly here :)